test/e2e/scrapeconfig_test.go

```diff
@@ -277,20 +277,11 @@ func testScrapeConfigKubernetesNodeRole(t *testing.T) {
 	_, err := framework.CreateOrUpdatePrometheusOperator(context.Background(), ns, []string{ns}, nil, []string{ns}, nil, false, true, true)
 	require.NoError(t, err)
 
-	p := framework.MakeBasicPrometheus(ns, "prom", "group", 1)
-	p.Spec.ScrapeConfigSelector = &metav1.LabelSelector{
-		MatchLabels: map[string]string{
-			"role": "scrapeconfig",
-		},
-	}
-	_, err = framework.CreatePrometheusAndWaitUntilReady(context.Background(), ns, p)
-	require.NoError(t, err)
```

Comment:
same remark here: it's way faster to create the Prometheus object after the scrape config.

Reply:
That sounds super obvious now that you mention it, but also super easy to miss when writing a test I think.

I wonder if we should add a section to TESTING.md like "e2e-test: tips and tricks" and mention how creating Prometheus/ThanosRuler last makes the test a lot faster